src/Analyzers/CSharp/Tests/UseIndexOrRangeOperator/UseRangeOperatorTests.cs

```diff
@@ -479,5 +478,64 @@ void Goo(Span<int> s)
                 FixedCode = fixedSource,
             }.RunAsync();
         }
+
+        [WorkItem(43202, "https://github.com/dotnet/roslyn/issues/43202")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseRangeOperator)]
+        public async Task TestWritableType()
+        {
+            var source =
+@"
+using System;
+struct S { 
+    public ref S Slice(int start, int length) => throw null; 
+    public int Length { get; } 
+    public S this[System.Range r] { get => default; } 
```

Comment:
💡 Also add a test just like this one, but with the argument to the indexer being `int` instead of `Range`. The refactoring should work in this case because the indexer syntax will continue to use the `Slice` method instead of the indexer with the wrong type.